dot/sync/service.go

```diff
@@ -88,12 +89,27 @@ type Change struct {
 type Strategy interface {
 	OnBlockAnnounce(from peer.ID, msg *network.BlockAnnounceMessage) (repChange *Change, err error)
 	OnBlockAnnounceHandshake(from peer.ID, msg *network.BlockAnnounceHandshake) error
-	NextActions() ([]*SyncTask, error)
-	Process(results []*SyncTaskResult) (done bool, repChanges []Change, blocks []peer.ID, err error)
+	NextActions() ([]Task, error)
+	Process(results <-chan TaskResult) (done bool, repChanges []Change, blocks []peer.ID, err error)
 	ShowMetrics()
 	IsSynced() bool
 }
 
+type syncTask struct {
+	requestMaker network.RequestMaker
+	request      messages.P2PMessage
+}
+
+func (s *syncTask) ID() TaskID {
+	return TaskID(s.request.String())
```

Comment:
I'm not sure if this is a good ID since it is the string representation of the request, if we send the same request for multiple peers we are gonna get the same ID here.
What if we just generate an UUID?
Also, we can add a `String()` for the string representation that could be useful for debugging purposes

Reply:
Good point! I had thought about whether a timestamp is a good ID for the batch, but hadn't considered this one. Just found the `String()` method and ran with it.
A UUID is a bit unwieldy, but we already use a library for generating them and I don't have a better idea, so went with that.
Also added `String()` and used it in logging calls.